src/shogun/multiclass/KNN.cpp

```diff
@@ -74,19 +74,20 @@ CKNN::~CKNN()
 
 bool CKNN::train_machine(CFeatures* data)
 {
-	ASSERT(m_labels)
-	ASSERT(distance)
+	REQUIRE(m_labels, "No training labels provided.\n");
+	REQUIRE(distance, "No training distance provided.\n");
 
 	if (data)
 	{
-		if (m_labels->get_num_labels() != data->get_num_vectors())
-			SG_ERROR("Number of training vectors does not match number of labels\n")
+		REQUIRE(
+		    m_labels->get_num_labels() == data->get_num_vectors(),
+		    "Number of training vectors does not match number of labels\n");
```

Comment:
give the numbers:
"Number of training data (%d) does not match number of labels (%d)\n"